pkg/services/hcloud/server/server.go

```diff
@@ -426,57 +893,97 @@ func (s *Service) reconcileLoadBalancerAttachment(ctx context.Context, server *h
 	return reconcile.Result{}, nil
 }
 
+func (s *Service) createServerFromImageNameOrURL(ctx context.Context) (*hcloud.Server, *hcloud.Image, error) {
+	if s.scope.HCloudMachine.Spec.ImageName != "" {
+		return s.createServerFromImageName(ctx)
+	}
+	return s.createServerFromImageURL(ctx)
+}
+
+func (s *Service) createServerFromImageURL(ctx context.Context) (*hcloud.Server, *hcloud.Image, error) {
+	// Validate that ImageURLCommand is given
+	hm := s.scope.HCloudMachine
+	if s.scope.ImageURLCommand == "" {
+		msg := fmt.Sprintf("controller was started without --hcloud-image-url-command. Provisioning via ImageURL %q not possible", hm.Spec.ImageURL)
+		conditions.MarkFalse(hm, infrav1.ServerAvailableCondition,
+			string(hm.Status.BootState), clusterv1.ConditionSeverityWarning,
+			"%s", msg)
+		return nil, nil, errors.New(msg)
+	}
+
+	image, err := s.getServerImage(ctx, preRescueOSImage)
+	if err != nil {
+		return nil, nil, fmt.Errorf("createServerFromImageURL: failed to get server image: %w", err)
+	}
+	server, err := s.createServer(ctx, s.scope.Name(), nil, image)
+	if err != nil {
+		return nil, nil, err
+	}
+	s.scope.HCloudMachine.SetBootState(infrav1.HCloudBootStateWaitForPreRescueOSThenEnableRescueSystem)
```

Comment:
you set this already outside of this function. I think you can delete that

Reply:
yes, I have seen that yesterday. Is fixed.